driver-core/src/main/com/mongodb/operation/AsyncChangeStreamBatchCursor.java

```diff
@@ -30,33 +33,54 @@
 import static com.mongodb.operation.ChangeStreamBatchCursorHelper.isRetryableError;
 import static com.mongodb.operation.OperationHelper.LOGGER;
 
-final class AsyncChangeStreamBatchCursor<T> implements AsyncBatchCursor<T> {
+final class AsyncChangeStreamBatchCursor<T> implements AsyncAggregateResponseBatchCursor<T> {
     private final AsyncReadBinding binding;
     private final ChangeStreamOperation<T> changeStreamOperation;
 
     private volatile BsonDocument resumeToken;
-    private volatile AsyncBatchCursor<RawBsonDocument> wrapped;
+    private volatile AsyncAggregateResponseBatchCursor<RawBsonDocument> wrapped;
 
     AsyncChangeStreamBatchCursor(final ChangeStreamOperation<T> changeStreamOperation,
-                                 final AsyncBatchCursor<RawBsonDocument> wrapped,
+                                 final AsyncAggregateResponseBatchCursor<RawBsonDocument> wrapped,
                                  final AsyncReadBinding binding) {
-        changeStreamOperation.startOperationTimeForResume(binding.getSessionContext().getOperationTime());
```

Comment:
Nit: Can this be updated mirror the sync version?  The ordering of setting vars etc. Abstracting checks to the wire version?

Keeping the code flow as similar as possible helps reduce the cognitive load between the sync / async codebases.